### PR TITLE
Add an ObservableSet to compliment List and Map

### PR DESCRIPTION
**Pull request by matanlurey**
## 0.20.0

* Add `ObservableSet`, `SetChangeRecord`, and `SetDiffer`

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,3 +1,7 @@
+## 0.20.0
```
Comment (natebosch) [inline CHANGELOG.md:1]: any breaking changes? should we call this 0.19.1?

**Review by natebosch — APPROVED** [review]

File: lib/src/collections/observable_map.dart [hunk 2]
```diff
@@ -13,7 +17,7 @@ import 'package:observable/observable.dart';
 ///     ```
 ///     set grades(Map<String, int> grades) {
 ///       buildBook(grades);
-///       if (names is ObservableMap<String>, int) {
+///       if (names is ObservableMap<String, int>) {
```
Comment (jmesserly) [inline lib/src/collections/observable_map.dart:20]: wow, nice catch!

**Review by jmesserly — APPROVED** [review]
LGTM!!! Yeah, fwiw, I don't see any breaking changes, so maybe release as 0.19.1.

sort of related: Filip H is always saying we should bump packages to 1.x whenever possible because we have a bad habit of leaving things at 0.x forever despite being code relied on in real apps. So that's another option whenever you feel ready

**Comment by natebosch** [comment]
I think it makes sense to hold off on 1.0 release while we're churning. Once we have these newer APIs nailed down we can probably go to 1.0 soon.

**Comment by jmesserly** [comment]
totally. just be careful, it tends to end up forever at 0.x. that's the cautionary tale from other pkgs :)

File: CHANGELOG.md (hunk 1 above)
Comment (matanlurey) [reply CHANGELOG.md:1]: _MINOR version when you add functionality in a backwards-compatible manner_

(http://semver.org/)

This adds functionality, so... :)

Comment (matanlurey) [reply CHANGELOG.md:1]: I plan on bumping to 1.0.0 as soon as the deprecations are all removed :)

**Comment by jmesserly** [comment]
FYI ... and this is a very minor point ... with 0.x versions, the rules of versioning are shifted to the right. So a functionality addition should be 0.19.1, a patch would be 0.19.0+1.

**Comment by jmesserly** [comment]
... and a breaking change would be 0.20

**Comment by jmesserly** [comment]
(and this is also why it's good not to stay at 0.x ... it leaves the version space pretty cramped to use semver and pub versions effectively)

**Pull request merged**